src/app/Http/Controllers/Admin/LaravelLogController.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Gate;
+
+class LaravelLogController extends Controller
+{
+    public function laravelLog(Request $request): JsonResponse|BinaryFileResponse
+    {
+        $user = $request->user();
```

Comment:
Permissions should be enforced using a policy. Look at app/Policies.

Reply:
Actually, I think policies only apply if you're dealing with models? It's worth looking into, but your code may be fine as written.

Reply:
Ya I looked into it, it only applies to models